packages/ata/src/index.ts

```diff
@@ -173,19 +173,21 @@ export const getReferencesForModule = (ts: typeof import("typescript"), code: st
     .filter(f => !isDtsFile(f.fileName))
     .filter(d => !libMap.has(d.fileName))
 
-  return references.map(r => {
-    let version = undefined
-    if (!r.fileName.startsWith(".")) {
-      version = "latest"
-      const line = code.slice(r.end).split("\n")[0]!
-      if (line.includes("// types:")) version = line.split("// types: ")[1]!.trim()
-    }
+  return references
+    .map(r => {
+      let version = undefined
+      if (!r.fileName.startsWith(".")) {
+        version = "latest"
+        const line = code.slice(r.end).split("\n")[0]!
+        if (line.includes("// types:")) version = line.split("// types: ")[1]!.trim()
+      }
 
-    return {
-      module: r.fileName,
-      version,
-    }
-  })
+      return {
+        module: r.fileName,
+        version,
+      }
+    })
+    .filter((r, index, self) => self.findIndex(m => m.module === r.module && m.version === r.version) === index)
```

Comment:
This is quadradic; how big are these inputs typically?

Reply:
I don't have a scientific answer, but I would think most files don't import more than on the order of 10s of unique packages. (I also don't think one would ever specify a different module versions for the same module? but just did the safest option for now, but happy to update) - I can also switch to a Set if preferred.

Reply:
Well, you had said there were 1000s of calls; does that occur without 1000s of entries in this list?

Reply:
Sorry, I updated my response (probably while you were responding, sorry about that), but I would not expect more than 10s of unique packages. So the array would not actually loop through 1000s of entries, but yes it can have 1000s.

Reply:
Ah, this is just in a single file. Gotcha.